iceoryx_posh/include/iceoryx_posh/internal/popo/untyped_subscriber.inl

```diff
@@ -26,6 +26,35 @@ inline UntypedSubscriberImpl<base_subscriber_t>::UntypedSubscriberImpl(const cap
 {
 }
 
+template <template <typename, typename, typename> class base_subscriber_t>
+inline cxx::expected<const void*, ChunkReceiveResult> UntypedSubscriberImpl<base_subscriber_t>::take_1_0() noexcept
+{
+    auto result = BaseSubscriber::takeChunk();
+    if (result.has_error())
+    {
+        return cxx::error<ChunkReceiveResult>(result.get_error());
+    }
+    return cxx::success<const void*>(result.value()->payload());
```

Comment:
payload should probably return a `const void*` when called from a `const ChunkHeader*`

Reply:
Then we need another overload for payload.

Reply:
oh right, this comes from the `ChunkHeader`. I'll add those overload with my feature. I also misread this as a cast, but it's actually just creating an `expected`.

Reply:
Yes, I did not want to introduce changes to Chunkheader and will hence do nothing here.